### PR TITLE
Move the `hir().krate()` method to a query and remove the `Krate` dep node

### PR DESCRIPTION
**Pull request by Zoxc**
r? @eddyb cc @michaelwoerister

### CODE REVIEW

**Review by michaelwoerister — COMMENTED** [review]
I haven't reviewed in detailed but looks good to me in general. I left a couple of comments below.

File: src/librustc/hir/mod.rs [hunk 1]
```diff
@@ -7,7 +7,48 @@ pub mod exports;
...
+    map: &'tcx map::Map<'tcx>,
+}
+
+impl<'tcx> Hir<'tcx> {
+    pub fn krate(&self) -> &'tcx Crate<'tcx> {
+        self.tcx.hir_crate(LOCAL_CRATE)
+    }
+}
+
+impl<'tcx> Deref for Hir<'tcx> {
```
Comment (eddyb) [inline src/librustc/hir/mod.rs]: This allows `tcx.hir().krate` from any code, and it's equivalent to `untracked_crate`, doubt that was intended.

Comment (Zoxc) [reply src/librustc/hir/mod.rs]: Yes, `krate` was supposed to be private.

**Review by eddyb — APPROVED** [review]
LGTM

**Pull request merged**